vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/implementation/CsvFileDataSource.java

```diff
@@ -22,103 +22,53 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.util.Arrays;
 
-import org.apache.commons.lang3.Validate;
 import org.eclipse.jdt.annotation.NonNull;
-import org.semanticweb.vlog4j.core.model.api.TermType;
-import org.semanticweb.vlog4j.core.reasoner.DataSource;
 
 /**
- * A CsvFileDataSource stores fact terms (tuples) as lines in a ".csv" format
- * file, each column being a predicate argument.
- * 
+ * A {@code CsvFileDataSource} stores facts as tuples inside a file of the
+ * extension {@code .csv}. One file can only include facts of a single
+ * predicate. The required format looks like this
```

Comment:
My comments are included in the review of #45

Reply:
I will incorporate all the updates to `new-extensions` into this branch with a rebase, once #45 was approved.